### ISSUE TITLE
julia_setup() deprecated Pkg use

### ISSUE THREAD

**Issue by matbesancon** [Non-Contradiction/JuliaCall#86]
On `julia_setup()` run:
```
running command ''/home/user/programs/julia/bin/julia' '/home/user/R/x86_64-pc-linux-gnu-library/3.4/JuliaCall/julia/install_dependency.jl' '/usr/lib/R' 2>&1' had status 1 
```

I think this is due to `Pkg.dir` is deprecated in https://github.com/Non-Contradiction/JuliaCall/blob/master/inst/julia/install_dependency.jl

**Comment by Non-Contradiction** [comment]
Thank you for the feedback.

The `Pkg.dir` may not be the reason for the status 1. But the status 1 definitely means something is not right...

The use of the deprecated `Pkg.dir` is intended. 
The purpose is to check whether rebuilding JuliaCall's dependency is needed or not,
and the new function which replaces the deprecated `Pkg.dir` is quite time-consuming. 
Hence I keep the `Pkg.dir`.

But after thinking the problem a bit more, since rebuilding is not common, which should only happen when R is updated to a new version, I may remove this checking in `julia_setup()`, and have a separate function like `rebuild()` which contains the checking and rebuilding, and users can use it only when needed. Then the new function `rebuild()` can use the more time-consuming checking function which is not deprecated.

**Comment by matbesancon** [comment]
it's a good idea yes, reading the readme, it's not clear what the function of setup() is and how often this should be performed

**Comment by Non-Contradiction** [comment]
The `julia_setup()` is necessary for each new R session to use the package. It will do all the necessary setup, like setting up the display system if necessary and loading `RCall.jl` in the embedded Julia, which is for type conversion between R and Julia, etc. And it will be invoked automatically before carrying out other `JuliaCall` functions. I will make this clear in the README.

I will also start a new development branch for the idea of `rebuild()` function, and it will be contained in the next release of `JuliaCall`.

**Comment by Non-Contradiction** [comment]
The deprecated Pkg usage has been removed in #88, and I add a new argument `rebuild` in the `julia_setup`  function, whose default value is `FALSE` and setting it to `TRUE` will trigger the rebuild of `RCall.jl` if necessary. 
I also update the documentation of `julia_setup` to make it more clear.
The pull request changes the functionality of the package, and it will be included in the next release.

**Comment by leouieda** [comment]
Linking to openjournals/joss-reviews#1284

**Comment by Non-Contradiction** [comment]
I merged #88 into master. After all, it only changes a little bit of functionality of the package, and I doubt any users will be negatively affected by this. So as far as I know, there is no deprecated usage of Julia now in the package.

**Comment by matbesancon** [comment]
ok so should we close this one?
